docs/developer/automerge.md

```diff
@@ -0,0 +1,25 @@
+# Automatic Merging of Pull Requests in OSM
+This document aims to explain how OSM utilizes the [automerge-action bot](https://github.com/pascalgn/automerge-action) to automatically merge pull requests and how it was set up.
+
+## Workaround to triggering workflows
+By default the automerge-action bot uses a `GITHUB_TOKEN` to perform automerging; however, events triggered via `GITHUB_TOKEN` will not further trigger additional workflow runs. To work around this, we created a GitHub App (Generate_Token) to generate a token that is then used to create a JSON Web Token (JWT) that authenticates and retrieves the Installation Access Token (IAT) from the GitHub API. Passing in the IAT as the `GITHUB_TOKEN`, will allow both `on:push` and `on:pull_request` workflows to trigger.
+
+## How it works
+- Automerge only occurs when the `automerge` label is assigned. To prevent automerging, you can either not assign the `automerge` label or include one of the following labels: `wip` or `do-not-merge/hold`.
+- Once labelled with `automerge` and neither of the blocking labels, a merge attempt will trigger only if all required status checks have passed (status checks are designated as required in the branch protection rules) and the required number of review approvals have been given.
```

Comment:
How does this work with `paths-ignore` (like a docs change)? wouldn't all the tests be required in all the PRs?

Reply:
you're right, all the status checks would have to be run for all PRs

Reply:
this is definitely a downside of this, it will eliminate any optional checks

Reply:
@sozercan just added a script that will wait for the status checks to complete so that branch protection rules will not need to be enabled